ackermann_steering_controller/include/ackermann_steering_controller/ackermann_steering_controller.h

```diff
@@ -0,0 +1,212 @@
+
+    /// Velocity command related:
+    struct Commands
+    {
+      double lin;
+      double ang;
+      ros::Time stamp;
+
+      Commands() : lin(0.0), ang(0.0), stamp(0.0) {}
+    };
+    realtime_tools::RealtimeBuffer<Commands> command_;
+    Commands command_struct_;
+    ros::Subscriber sub_command_;
+
+    /// Odometry related:
+    boost::shared_ptr<realtime_tools::RealtimePublisher<nav_msgs::Odometry> > odom_pub_;
+    boost::shared_ptr<realtime_tools::RealtimePublisher<tf::tfMessage> > tf_odom_pub_;
+    Odometry odometry_;
+
+    /// Wheel separation, wrt the midpoint of the wheel width:
+    double wheel_separation_h_;
+
+    /// Wheel radius (assuming it's the same for the left and right wheels):
+    double wheel_radius_;
+
+    /// Wheel separation and radius calibration multipliers:
+    double wheel_separation_h_multiplier_;
+    double wheel_radius_multiplier_;
+    double steer_pos_multiplier_;
+
+    /// Timeout to consider cmd_vel commands old:
+    double cmd_vel_timeout_;
+
+    /// Whether to allow multiple publishers on cmd_vel topic or not:
+    bool allow_multiple_cmd_vel_publishers_;
+
+    /// Frame to use for the robot base:
+    std::string base_frame_id_;
+
+    /// Frame to use for odometry and odom tf:
+    std::string odom_frame_id_;
+
+    /// Whether to publish odometry to tf or not:
+    bool enable_odom_tf_;
+
+    /// Number of wheel joints:
+    size_t wheel_joints_size_;
+
+    /// Number of steer joints:
+    size_t steer_joints_size_;
+
+    /// Speed limiters:
+    Commands last1_cmd_;
+    Commands last0_cmd_;
+    SpeedLimiter limiter_lin_;
+    SpeedLimiter limiter_ang_;
+
+    // FOR_DEBUG
+    std::string ns_;
+
+  private:
+    /**
+     * \brief Brakes the wheels, i.e. sets the velocity to 0
+     */
+    void brake();
+
+    /**
+     * \brief Velocity command callback
+     * \param command Velocity command message (twist)
+     */
+    void cmdVelCallback(const geometry_msgs::Twist& command);
+
+    /**
+     * \brief Sets odometry parameters from the URDF, i.e. the wheel radius and separation
+     * \param root_nh Root node handle
+     * \param left_wheel_name Name of the left wheel joint
+     * \param right_wheel_name Name of the right wheel joint
+     */
+    bool setOdomParamsFromUrdf(ros::NodeHandle& root_nh,
+                               const std::string rear_wheel_name,
+                               const std::string front_steer_name,
+                               bool lookup_wheel_separation_h,
+                               bool lookup_wheel_radius);
+
+    /**
+     * \brief Sets the odometry publishing fields
+     * \param root_nh Root node handle
+     * \param controller_nh Node handle inside the controller namespace
+     */
+    void setOdomPubFields(ros::NodeHandle& root_nh, ros::NodeHandle& controller_nh);
+
+  };
+
+  PLUGINLIB_EXPORT_CLASS(ackermann_steering_controller::AckermannSteeringController, controller_interface::ControllerBase)
```

Comment:
this must be moved into a .cpp file.

Reply:
Fixed at https://github.com/ros-controls/ros_controllers/pull/356/commits/be7d149661b978313a123a46c4539b80f9d4909f